File: server/src/B1ngo.Application.Features/Rooms/GetRoomState/GetRoomStateResponse.cs
namespace B1ngo.Application.Features.Rooms.GetRoomState;

public sealed record GetRoomStateResponse(
    Guid RoomId,
    string JoinCode,
    string Status,
    SessionDto Session,
    ConfigurationDto Configuration,
    Guid HostPlayerId,
    IReadOnlyList<PlayerDto> Players,
    IReadOnlyList<LeaderboardEntryDto> Leaderboard
);

public sealed record SessionDto(int Season, string GrandPrixName, string SessionType);

public sealed record ConfigurationDto(int MatrixSize, IReadOnlyList<string> WinningPatterns);

public sealed record PlayerDto(Guid PlayerId, string DisplayName, bool HasWon, CardDto? Card);

public sealed record CardDto(int MatrixSize, IReadOnlyList<SquareDto> Squares);

public sealed record SquareDto(
    int Row,
    int Column,
    string DisplayText,
    string? EventKey,
    bool IsFreeSpace,
    bool IsMarked,
    string? MarkedBy,
    DateTimeOffset? MarkedAt
);

public sealed record LeaderboardEntryDto(Guid PlayerId, int Rank, string WinningPattern, DateTimeOffset CompletedAt);

File: client/apps/b1ngo/src/app/core/api/models/responses/get-room-state.response.ts
import { RoomStatus } from '../types/room-status.type';
import { SessionDto } from '../dtos/session.dto';
import { ConfigurationDto } from '../dtos/configuration.dto';
import { PlayerDto } from '../dtos/player.dto';
import { LeaderboardEntryDto } from '../dtos/leaderboard-entry.dto';

export interface GetRoomStateResponse {
  roomId: string;
  currentPlayerId: string;
  joinCode: string;
  status: RoomStatus;
  session: SessionDto;
  configuration: ConfigurationDto;
  hostPlayerId: string;
  players: PlayerDto[];
  leaderboard: LeaderboardEntryDto[];
}
